ui_test/src/lib.rs

```diff
@@ -94,11 +96,50 @@ pub fn run_tests(config: Config) -> Result<()> {
             drop(submit);
         });
 
+        // A channel for the messages emitted by the individual test threads.
+        let (finish_file, finished_files) = crossbeam::channel::unbounded();
```

Comment:
I think I'd prefer names that indicate which is the sender and which the receiver (_send/_recv suffices, or something like that).